File: Assets/Scripts/datastructs/Command.cs
using System.Linq;

public enum CommandType : byte {
    CANCEL = 0,
    EXIT = 1,
}

public class Command {
    public CommandType type;

    public Command(CommandType ct) {
        type = ct;
    }

    public byte[] ToBytes() {
        return new byte[] { (byte)type };
    }

    public override string ToString() {
        return $"Command: {type}";
    }
}

public class DirectCommand : Command {
    public string target;

    public DirectCommand(CommandType ct, string target) : base(ct) {
        this.target = target;
    }

    public override string ToString() {
        return $"DirectCommand: {type} for {target}";
    }
}

File: Assets/Python/root/datastructs/command.py
from enum import IntEnum
import struct

class CommandType(IntEnum):
    CANCEL = 0
    EXIT = 1

class Command:
    def __init__(self, ct: CommandType):
        self.type = ct
    def to_bytes(self):
        return struct.pack('!B', self.type.value)
    def __str__(self) -> str:
        return "CANCEL" if self.type == CommandType.CANCEL else "EXIT"

class DirectCommand(Command):
    def __init__(self, ct: CommandType, target: str):
        Command.__init__(self, ct)
        self.target = target

    def __str__(self) -> str:
        return super().__str__() + f" for {self.target}"